File: task-03/sub-task02/Ruby/readandwrite.rb

input_file_path = 'input.txt'
output_file_path = 'output.txt'

begin
  #Reading 
  data = File.read(input_file_path)
  
  # Writing 
  File.write(output_file_path, data)

  # Reading from the output file and printing
  output_data = File.read(output_file_path)
  puts 'Contents of output.txt'
  puts output_data

rescue Errno::ENOENT => e
  # Handle file not found errors
  puts 'Error Input '
  puts e.message
rescue IOError => e
  # Handle other I/O errors
  puts 'Error  I/O '
  puts e.message
end

File: task-03/sub-task02/Elixir/readandwrite.exs
defmodule FileProcessor do
  @input_file_path "input.txt"
  @output_file_path "output.txt"

  def run do
    case File.read(@input_file_path) do
      {:ok, data} ->
        case File.write(@output_file_path, data) do
          :ok ->
            case File.read(@output_file_path) do
              {:ok, output_data} ->
                IO.puts "Contents of output.txt:"
                IO.puts output_data

              {:error, reason} ->
                IO.puts "Error reading from output file."
                IO.puts reason
            end

          {:error, reason} ->
            IO.puts "Error writing to output file."
            IO.puts reason
        end

      {:error, reason} ->
        IO.puts "Error reading from input file."
        IO.puts reason
    end
  end
end


